python-frontend/src/main/java/org/sonar/python/tree/NumericLiteralImpl.java

```diff
@@ -21,19 +21,24 @@
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 import org.sonar.plugins.python.api.tree.NumericLiteral;
 import org.sonar.plugins.python.api.tree.Token;
-import org.sonar.plugins.python.api.tree.TreeVisitor;
 import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.TreeVisitor;
+import org.sonar.plugins.python.api.types.InferredType;
+import org.sonar.python.types.InferredTypes;
 
 public class NumericLiteralImpl extends PyTree implements NumericLiteral {
 
   private final String valueAsString;
+  private final String valueAsStringLowerCase;
```

Comment:
Why do we store that as a field? If we want to store something as a field, I would prefer to store the inferred type.